src/OrchardCore.Modules/OrchardCore.Media/Assets/js/Shared/uploadComponent.js

```diff
@@ -11,18 +11,20 @@ Vue.component('upload', {
         </div> \
         ',
     props: {
-        model: Object
+        model: Object,
+        idOfUploadInputField: String
```

Comment:
Needless to say this it too long name that translates as attribute `id-of-upload-input-field`, 

I guess this prop is not needed and use `id` instead, and you can always bind  to `:id` in vue

Reply:
I agree that the name is too long. I only used it because it was already specified here: https://github.com/OrchardCMS/OrchardCore/blob/b9c2d5122976b5dcfaf0263a5da643a7cd6d7e82/src/OrchardCore.Modules/OrchardCore.Media/Views/MediaField-Attached.Edit.cshtml#L43
I don't understand how it is not needed though. It seems like to me there needs to be a property so the Id of the file input element can be passed to the components since that element is not part of the component.

Reply:
We are free to update it for the better

Reply:
What do you think of changing it to `uploadInputId` or do you have suggestions on a good name?

Reply:
id seems unnecessary, use `ref` instead to refer element within vue template

Reply:
@ns8482e I'm not very familiar with Vue.js so I might be misunderstanding you but I read the [Vue documentation for `ref`](https://vuejs.org/guide/essentials/template-refs.html#accessing-the-refs) and don't see how that would allow me to access the element in this case.  The reason I think it needs the file input id passed as a param to the component is because the file input element is not part of the vue component. It is in the orchard core views for the attached media editor and the media library application.
https://github.com/OrchardCMS/OrchardCore/blob/b9c2d5122976b5dcfaf0263a5da643a7cd6d7e82/src/OrchardCore.Modules/OrchardCore.Media/Views/MediaField-Attached.Edit.cshtml#L31
https://github.com/OrchardCMS/OrchardCore/blob/b9c2d5122976b5dcfaf0263a5da643a7cd6d7e82/src/OrchardCore.Modules/OrchardCore.Media/Views/Admin/MediaApplication.cshtml#L69

Reply:
Renamed id-of-upload-input-field to upload-input-id. https://github.com/OrchardCMS/OrchardCore/pull/12956/commits/4de1b7946d393479377f3c82d2be203517b68750

Reply:
@DrewBrasher oh ok. - so, here the scenario is the element is outside of the vue component and trying to handle an event by vue component?

Reply:
@ns8482e Yes